internal/task/cgroups/cgroup.go

```diff
@@ -0,0 +1,102 @@
+package cgroups
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"syscall"
+	"time"
+)
+
+const baseCgroupPath = "/sys/fs/cgroup/"
+
+// CreateCgroupForTask creates a cgroup for a task
+// In a real system, we would want the cgroup config to be configurable
+func CreateCgroupForTask(taskID string) (*os.File, error) {
```

Comment:
Are we sure the system has the cgroup controllers enabled?

Reply:
I'll add a check on server start up.